llama_index/vector_stores/qdrant.py

```diff
@@ -347,16 +420,60 @@ def query(
         Args:
             query (VectorStoreQuery): query
         """
+        from qdrant_client import models as rest
         from qdrant_client.http.models import Filter
 
         query_embedding = cast(List[float], query.query_embedding)
+        query_filter = cast(Filter, self._build_query_filter(query))
 
-        response = self._client.search(
-            collection_name=self.collection_name,
-            query_vector=query_embedding,
-            limit=cast(int, query.similarity_top_k),
-            query_filter=cast(Filter, self._build_query_filter(query)),
-        )
+        if query.mode == VectorStoreQueryMode.HYBRID and not self.enable_hybrid:
+            raise ValueError(
+                "Hybrid search is not enabled. Please build the query with "
+                "`enable_hybrid=True` in the constructor."
+            )
+        elif query.mode == VectorStoreQueryMode.HYBRID and self.enable_hybrid:
+            sparse_indices, sparse_embedding = self._sparse_query_fn(
+                [query.query_str],
+            )
+            sparse_top_k = query.sparse_top_k or query.similarity_top_k
+
+            sparse_response = self._client.search_batch(
+                collection_name=self.collection_name,
+                requests=[
+                    rest.SearchRequest(
+                        vector=rest.NamedVector(
+                            name="text-dense",
+                            vector=query_embedding,
+                        ),
+                        limit=query.similarity_top_k,
+                        filter=query_filter,
+                        with_payload=True,
+                    ),
+                    rest.SearchRequest(
+                        vector=rest.NamedSparseVector(
+                            name="text-sparse",
+                            vector=rest.SparseVector(
+                                indices=sparse_indices[0],
+                                values=sparse_embedding[0],
+                            ),
+                        ),
+                        limit=sparse_top_k,
+                        filter=query_filter,
+                        with_payload=True,
+                    ),
+                ],
+            )
+            assert len(sparse_response) == 2  # sanity check
+
+            # flatten the response
+            response = sparse_response[0] + sparse_response[1]
```

Comment:
I can add `alpha` here for merging if this is the place

Reply:
The sparse results have quite a different `score` than the dense results.

Probably, our `hybrid_fusion_fn()` should be handling any kind of alpha or other approaches. Let me refactor this slightly

Reply:
Refactored :)

Reply:
and added alpha support 👍🏻